openwebstart/src/main/java/com/openwebstart/jvm/vendor/VendorManager.java

```diff
@@ -0,0 +1,40 @@
+package com.openwebstart.jvm.vendor;
+
+import net.adoptopenjdk.icedteaweb.Assert;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.ServiceLoader;
+
+public class VendorManager {
+
+    private final static String UNKNOWN_VENDOR = "Unknown vendor";
+
+    private final static VendorManager INSTANCE = new VendorManager();
+
+    private final List<VendorResolver> resolver;
+
+    public VendorManager() {
+        final List<VendorResolver> loaded = new ArrayList<>();
+        ServiceLoader.load(VendorResolver.class).iterator().forEachRemaining(r -> loaded.add(r));
+        resolver = Collections.unmodifiableList(loaded);
+    }
+
+    public String getInternalName(final String name) {
```

Comment:
if this method would be able to handle `null`, `emptyString` and `RuntimeManagerConstants.VENDOR_ANY` correctly a lot of boiler plate code in the clients of this method could be removed